src/util/util.ts

```diff
@@ -35,3 +35,8 @@ export const taskLogger =
 
 export const isSubstringOf = (substring: string, source: string): boolean =>
   pipe(source, S.toLowerCase, S.includes(S.toLowerCase(substring)))
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const isError = (value: any): value is Error =>
+  value.name === 'Error' &&
+  Object.keys(value).every(key => ['cause', 'stack', 'message'].includes(key))
```

Comment:
Checking for the name property to be exactly 'Error' is risky as it may lead to false negatives for Error subtypes.
Also, not all errors have a `cause` property, and custom error classes might not have these properties.

I would directly add the `instanceof` check within the component.

If you still want to create a typeguard function, here is my suggestion:
```suggestion
export const isError = (value: unknown): value is Error =>
  value instanceof Error;
```

Reply:
Good point. I imagine this guard function can be useful in a general sense as catch statement arguments are unknown.